lib/internal/http.js

```diff
@@ -132,6 +132,13 @@ class ProxyConfig {
     };
   }
 
+  // Private helper method for performance optimization
```

Comment:
I don't think the helper would make a difference in performance? (FWIW I implemented & optimized private methods in V8 a few years ago and the last time I looked into it, V8 should not treat this differently)